src/ConstructiveSolidGeometry/ConstructiveSolidGeometry.jl

```diff
@@ -64,10 +67,15 @@ module ConstructiveSolidGeometry
     _csg_get_promoted_eltype(::Type{Tuple{T1,T2}}) where {T1<:Union{Real, Tuple}, T2<:Union{Real, Tuple}} = promote_type(_csg_get_promoted_eltype(T1), _csg_get_promoted_eltype(T2))
     _csg_get_promoted_eltype(::Type{Tuple{Nothing,T2}}) where {T2<:Union{Real, Tuple}} = _csg_get_promoted_eltype(T2)
     _csg_get_promoted_eltype(::Type{Tuple{T1,Nothing}}) where {T1<:Union{Real, Tuple}} = _csg_get_promoted_eltype(T1)
-    _csg_get_promoted_eltype(::Type{Quan}) where {Quan<:Quantity}= Quan.parameters[1]
+    _csg_get_promoted_eltype(::Type{Q}) where {Q <: Quantity}= Q
     
     _handle_phi(φ, rotation) = (φ, rotation)
     _handle_phi(φ::Tuple, rotation) = (abs(φ[2]-φ[1]), rotation*RotZ(φ[1]))
+    
+    _precision_type(::Type{T}) where {T <: Real} = T
+    _precision_type(::Type{Quantity{T}}) where {T <: Real} = T
+    
+    float(::Type{Quantity{T}}) where T = T
```

Comment:
Careful! This is type piracy!
`Quantity` is a type that SSD does not own, so we should be **VERY** careful when extending the `Base.float` method